test/runtests.jl

```diff
@@ -209,6 +209,13 @@ float_types = [Float32, Float64]
                 @test poisson_ppn_planned_div_free_cpu(ft, Nx, Ny, Nz, FFTW.ESTIMATE)
             end
         end
+
+        @testset "Analytic solution reconstruction" begin
+            println("    Testing analytic solution reconstruction...")
+            for N in [32, 48, 64], m in [1, 2, 3]
+                @test poisson_ppn_recover_sine_cosine_solution(Float64, N, N, N, 100, 100, 100, m, m, m)
```

Comment:
Any reason for not using just 1, 1, 1 for the length?

Does it make sense to provide multiple error tolerances if we are testing more than one resolution? Seems like we don't get much out of testing the lower resolutions here: if the higher resolutions pass, the lower ones are basically guaranteed to.

Reply:
I take back my suggestion for 1, 1, 1 --- you should definitely use 2π, 2π, π for simplicity!

Reply:
Not really, (1, 1, 1) makes more sense. I just usually default to 100, 100, 100 I guess since most of the simulations I've done were on that domain lol.

I agree it would be good to have multiple error tolerances.

The reason I did multiple resolutions and logged the error is just to see if the error decreases as expected. I'm going to start doing this more often so that we get some useful debug/diagnostic information from the test suite.